src/children.js

```diff
@@ -29,3 +31,27 @@ export const properties = children.properties = {
 
 // Old JSEP versions
 properties.LogicalExpression = properties.BinaryExpression;
+
+/**
+ * Replace a child node with a new node by updating its parent
+ * @param {object} child the node to replace
+ * @param {object} newChild the replacement node
+ */
+export function replace (child, newChild) {
+	const parent = parents.get(child);
+	if (!parent) {
+		return;
+	}
```

Comment:
Not sure either, but I'm leaning the same way as you. E.g. it may be the root, then it's normal to not have a parent (though there the parent should be `null`, not `undefined`.